test/snapshots/no-array-method-this-argument.mjs.md

```diff
@@ -0,0 +1,379 @@
+        |                      ^^^^^^^^^^^^ Do not use this argument in \`Array#find()\`.␊
+    `
+
+## Invalid #4
+      1 | array.findIndex(() => {}, thisArgument)
+
+> Output
+
+    `␊
+      1 | array.findIndex(() => {})␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | array.findIndex(() => {}, thisArgument)␊
+        |                           ^^^^^^^^^^^^ Do not use this argument in \`Array#findIndex()\`.␊
+    `
+
+## Invalid #5
+      1 | array.flatMap(() => {}, thisArgument)
+
+> Output
+
+    `␊
+      1 | array.flatMap(() => {})␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | array.flatMap(() => {}, thisArgument)␊
+        |                         ^^^^^^^^^^^^ Do not use this argument in \`Array#flatMap()\`.␊
+    `
+
+## Invalid #6
+      1 | array.forEach(() => {}, thisArgument)
+
+> Output
+
+    `␊
+      1 | array.forEach(() => {})␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | array.forEach(() => {}, thisArgument)␊
+        |                         ^^^^^^^^^^^^ Do not use this argument in \`Array#forEach()\`.␊
+    `
+
+## Invalid #7
+      1 | array.map(() => {}, thisArgument)
+
+> Output
+
+    `␊
+      1 | array.map(() => {})␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | array.map(() => {}, thisArgument)␊
+        |                     ^^^^^^^^^^^^ Do not use this argument in \`Array#map()\`.␊
+    `
+
+## Invalid #8
+      1 | array.map(() => {}, thisArgument,)
+
+> Output
+
+    `␊
+      1 | array.map(() => {},)␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | array.map(() => {}, thisArgument,)␊
+        |                     ^^^^^^^^^^^^ Do not use this argument in \`Array#map()\`.␊
+    `
+
+## Invalid #9
+      1 | array.map(() => {}, (0, thisArgument),)
+
+> Output
+
+    `␊
+      1 | array.map(() => {},)␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | array.map(() => {}, (0, thisArgument),)␊
+        |                      ^^^^^^^^^^^^^^^ Do not use this argument in \`Array#map()\`.␊
+    `
+
+## Invalid #10
+      1 | array.map(() => {}, thisArgumentHasSideEffect())
+
+> Error 1/1
+
+    `␊
+    > 1 | array.map(() => {}, thisArgumentHasSideEffect())␊
+        |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not use this argument in \`Array#map()\`.␊
+    ␊
+    --------------------------------------------------------------------------------␊
+    Suggestion 1/1: Remove the second argument.␊
+      1 | array.map(() => {})␊
+    `
+
+## Invalid #1
+      1 | array.map(callback, thisArgument)
+
+> Error 1/1
+
+    `␊
+    > 1 | array.map(callback, thisArgument)␊
+        |                     ^^^^^^^^^^^^ Do not use this argument in \`Array#map()\`.␊
+    ␊
+    --------------------------------------------------------------------------------␊
+    Suggestion 1/2: Remove the second argument.␊
+      1 | array.map(callback)␊
+    ␊
+    --------------------------------------------------------------------------------␊
+    Suggestion 2/2: Use bound function.␊
+      1 | array.map(callback.bind(thisArgument))␊
```

Comment:
I didn't auto-fix this case to use `callback.bind`, since it's not the best option, people may won't get attention with auto-fix.